var/spack/repos/builtin/packages/quda/package.py

```diff
@@ -0,0 +1,193 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Quda(CMakePackage, CudaPackage, ROCmPackage):
+    """QUDA is a library for performing calculations in lattice QCD on GPUs."""
+
+    homepage = "https://lattice.github.io/quda/"
+    url = "https://github.com/lattice/quda/archive/refs/tags/v1.1.0.tar.gz"
+    git = "https://github.com/lattice/quda.git"
+
+    maintainers("chaoos")
+    license("MIT OR BSD-3-Clause", checked_by="chaoos")
+
+    version("develop", branch="develop")
+    version(
+        "experimental",
+        branch="feature/openqxd",
+        git="https://github.com/chaoos/quda.git",
+        get_full_repo=True,
+    )  # todo: change to release
+    version("2025-01-23", preferred=True, commit="18bf43ed40c75ae276e55bb8ddf2f64aa5510c37")
+    version("1.1.0", sha256="b4f635c993275010780ea09d8e593e0713a6ca1af1db6cc86c64518714fcc745")
+    version(
+        "1.0.0",
+        deprecated=True,
+        sha256="32b883bd4af45b76a832d8a070ab020306c94ff6590410cbe7c3eab3b630b938",
+    )
+    version(
+        "0.9.0",
+        deprecated=True,
+        sha256="0a9f2e028fb40e4a09f78af51702d2de4099a9bf10fb8ce350eacb2d8327e481",
+    )
+    version(
+        "0.8.0",
+        deprecated=True,
+        sha256="58d9a94b7fd38ec1f79bea7a0e9b470f0fa517cbf5fce42d5776e1c65607aeda",
+    )
+
+    # build dependencies
+    generator("ninja")
+    depends_on("cmake@3.24", type="build")
```

Comment:
This likely needs to be an open-ended range:
```suggestion
    depends_on("cmake@3.24:", type="build")
```
I also can't confirm the minimum version, [which is 3.18 in the upstream project](https://github.com/lattice/quda/blob/develop/CMakeLists.txt#L16).

Reply:
Changed to `@3.18:`.